File: ai/src/main/java/org/pingel/causality/examples/Model3dot9a.java
package org.pingel.causality.examples;

import org.pingel.bayes.CausalModel;
import org.pingel.bayes.Function;
import org.pingel.bayes.ModelVisualizer;
import org.pingel.bayes.RandomVariable;


public class Model3dot9a extends CausalModel
{
    public Model3dot9a()
    {
        super("3.9a");

        RandomVariable X = addVariable(new RandomVariable("X", "x"));
		RandomVariable Y = addVariable(new RandomVariable("Y", "y"));
		RandomVariable U1 = addVariable(new RandomVariable("U1", "u1", false));

		addFunction(new Function(X, U1));
		addFunction(new Function(Y, X, U1));
    }
    
    public static void main(String[] argv)
    {
        CausalModel model = new Model3dot9a();
        ModelVisualizer.draw(model);
    }
    
}

File: ai/src/main/scala/org/pingel/causality/examples/Model3dot9a.scala
package org.pingel.causality.examples

import org.pingel.causality.CausalModel
import org.pingel.causality.Function
import org.pingel.bayes.RandomVariable

object Model3dot9a extends CausalModel("3.9a") {

  val X = new RandomVariable("X")
  val Y = new RandomVariable("Y")
  val U1 = new RandomVariable("U1", None, false)

  g ++= (X :: Y :: U1 :: Nil)
  
  addFunction(new Function(X, List(U1)))
  addFunction(new Function(Y, List(X, U1)))

  def main(args: Array[String]) {
    g.draw
  }

}
